kafka-connect-s3/src/test/java/io/confluent/connect/s3/integration/BaseConnectorIT.java

```diff
@@ -0,0 +1,483 @@
+  protected static final String CONNECTOR_NAME = "s3-sink";
+  protected static final int MAX_TASKS = 3;
+
+  private static final Logger log = LoggerFactory.getLogger(BaseConnectorIT.class);
+  private static final String TEST_RESOURCES_PATH = "src/test/resources/";
+  private static final String AWS_REGION = "us-west-2";
+  private static final String TEST_DOWNLOAD_PATH = TEST_RESOURCES_PATH + "downloaded-files/";
+  private static final long CONNECTOR_STARTUP_DURATION_MS = TimeUnit.MINUTES.toMillis(1);
+  private static final long S3_TIMEOUT_MS = TimeUnit.SECONDS.toMillis(30);
+
+  protected static AmazonS3 S3Client;
+  protected EmbeddedConnectCluster connect;
+  protected Map<String, String> props;
+  private ObjectMapper jsonMapper = new ObjectMapper();
+
+  protected static boolean useMockClient() {
+    File creds = new File(AWS_CRED_PATH);
+    return System.getenv(JENKINS_HOME) != null || !creds.exists();
+  }
+
+  /**
+   * Get an S3 client based on existing credentials, or a mock client if running on jenkins.
+   *
+   * @return an authenticated S3 client
+   */
+  private static AmazonS3 getS3Client() {
+    if (useMockClient()) {
+      // S3Mock.create(8001, "/tmp/s3");
+      S3Mock api = new S3Mock.Builder().withPort(MOCK_S3_PORT).withInMemoryBackend().build();
+      api.start();
+      /*
+       * AWS S3 client setup.
+       * withPathStyleAccessEnabled(true) trick is required to overcome S3 default
+       * DNS-based bucket access scheme
+       * resulting in attempts to connect to addresses like "bucketname.localhost"
+       * which requires specific DNS setup.
+       */
+      EndpointConfiguration endpoint = new EndpointConfiguration(MOCK_S3_URL, AWS_REGION);
+      AmazonS3 mockClient = AmazonS3ClientBuilder
+          .standard()
+          .withPathStyleAccessEnabled(true)
+          .withEndpointConfiguration(endpoint)
+          .withCredentials(new AWSStaticCredentialsProvider(new AnonymousAWSCredentials()))
+          .build();
+
+      mockClient.createBucket(TEST_BUCKET_NAME);
+      log.info("No credentials found, using mock S3 client.");
+      return mockClient;
+    } else {
+      log.info("Credentials found, using real S3 client.");
+      // DefaultAWSCredentialsProviderChain,
+      // assumes .aws/credentials is setup and test bucket exists
+      return AmazonS3ClientBuilder.standard().withRegion(AWS_REGION).build();
+    }
+  }
+
+  @BeforeClass
+  public static void setupClient(){
+    S3Client = getS3Client();
+  }
+
+  @Before
+  public void setup() throws ConnectException {
+    startConnect();
+    setupProperties();
+    // clear in case of an existing bucket
+    clearBucket(TEST_BUCKET_NAME);
+  }
+
+  @After
+  public void close() throws Throwable {
+    // stop all Connect, Kafka and Zk threads.
+    connect.stop();
+    // delete the downloaded test file folder
+    FileUtils.deleteDirectory(new File(TEST_DOWNLOAD_PATH));
+    clearBucket(TEST_BUCKET_NAME);
```

Comment:
maybe we should delete the buckets esp if we are using actual S3. should fully clean up the test environment